src/categories/categories.service.ts

```diff
@@ -58,17 +64,38 @@ export class CategoriesService {
     }
   }
 
-  async getAllApis(categoryId: string) {
+  async getAllApis(
+    categoryId: string,
+    query: PaginateQuery,
+  ): Promise<Paginated<Api>> {
     try {
       const category = await this.categoryRepo.findOne({
         where: { id: categoryId },
       });
+      
       if (category.name.toLowerCase() == 'all')
-        return await this.apiRepo.find({
+        return paginate(query, this.apiRepo, {
+          sortableColumns: ['createdOn', 'name', 'visibility'],
+          searchableColumns: ['name', 'description', 'about', 'visibility'],
+          defaultSortBy: [['id', 'DESC']],
           where: { visibility: Visibility.Public },
+          filterableColumns: {
+            category: [FilterOperator.IN],
+            status: [FilterOperator.IN],
+            rating: [FilterOperator.GTE, FilterOperator.LTE],
+          },
         });
```

Comment:
What is the difference between the two  conditions with respect to category.name === 'all'?, the conditions are the same. You are essentially writing the same code for no reason.